### PR TITLE
[zh-cn] init translation of `BarcodeDetector()`

### PR DESCRIPTION
**Pull request by skyclouds2001**
### Description

https://developer.mozilla.org/en-US/docs/Web/API/BarcodeDetector/BarcodeDetector

### Related issues and pull requests

Depends on: https://github.com/mdn/content/pull/31115

### CODE REVIEW

File: files/zh-cn/web/api/barcodedetector/barcodedetector/index.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+```
+
+### 参数
+
+- `options` {{optional_inline}}
+
+  - : 包含一系列 `BarcodeFormats` 的选项对象，用于在后续调用 {{domxref('BarcodeDetector.detect()','detect()')}} 中搜索。选项有：
+
+    - `formats` {{optional_inline}}
+      - : 一个表示条形码格式的字符串的 {{jsxref('Array')}}。 如果未提供，调用 `detect()` 将搜索所有支持的格式。因此，出于性能原因，建议限制为具体的格式。要查看支持格式的完整列表，请参阅 {{domxref("BarcodeDetection API")}}。
```
Comment (skyclouds2001) [inline files/zh-cn/web/api/barcodedetector/barcodedetector/index.md:26]:
```suggestion
      - : 一个表示条形码格式的字符串的 {{jsxref('Array')}}。如果未提供，调用 `detect()` 方法将搜索所有支持的格式。因此，出于性能原因，建议限制为具体的格式。要查看支持格式的完整列表，请参阅 {{domxref("BarcodeDetection API")}}。
```

File: files/zh-cn/web/api/barcodedetector/barcodedetector/index.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+### 异常
+
+- {{jsxref("TypeError")}}
+  - : Thrown if the `formats` is specified and the parameter is empty or contains `unknown`.
```
Comment (skyclouds2001) [inline files/zh-cn/web/api/barcodedetector/barcodedetector/index.md:31]:
```suggestion
  - : 如果指定了“formats”参数并且该参数为空数组或包含“unknown”，则抛出此异常。
```

File: files/zh-cn/web/api/barcodedetector/barcodedetector/index.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+```
+
+### 参数
+
+- `options` {{optional_inline}}
+
+  - : 包含一系列 `BarcodeFormats` 的选项对象，用于在后续调用 {{domxref('BarcodeDetector.detect()','detect()')}} 中搜索。选项有：
+
+    - `formats` {{optional_inline}}
+      - : 一个表示条形码格式的字符串 {{jsxref('Array')}}。如果未提供，调用 `detect()` 方法时将搜索所有支持的格式。因此，出于性能原因，建议限制为具体的格式。要查看支持格式的完整列表，请参阅 {{domxref("Barcode Detection API")}}。
```
Comment (skyclouds2001) [inline files/zh-cn/web/api/barcodedetector/barcodedetector/index.md:26]:
```suggestion
      - : 一个表示条形码格式的字符串 {{jsxref('Array')}}。如果未提供，调用 `detect()` 方法时将搜索所有支持的格式。因此，出于性能原因，建议限制为具体的格式。要查看支持格式的完整列表，请参阅 {{domxref("Barcode Detection API", "", "", "nocode")}}。
```

File: files/zh-cn/web/api/barcodedetector/barcodedetector/index.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
+---
+title: BarcodeDetector：BarcodeDetector() 构造方法
+slug: Web/API/BarcodeDetector/BarcodeDetector
+l10n:
+  sourceCommit: 236d6a07df44912cfd6d55d33bd8869fad173040
+---
+
+{{securecontext_header}}{{APIRef("Barcode Detector API")}}{{AvailableInWorkers}}{{SeeCompatTable}}
+
+**`BarcodeDetector()`** 构造函数创建一个新的 {{domxref("BarcodeDetector")}} 对象，用于检测图像中条形码和二维码。
```
Comment (yin1999) [inline files/zh-cn/web/api/barcodedetector/barcodedetector/index.md:10]: How about:

```md
创建一个新的、用于检测图像中条形码和二维码的 {{domxref("BarcodeDetector")}} 对象。
```

File: files/zh-cn/web/api/barcodedetector/barcodedetector/index.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
+---
+title: BarcodeDetector：BarcodeDetector() 构造方法
+slug: Web/API/BarcodeDetector/BarcodeDetector
+l10n:
+  sourceCommit: 236d6a07df44912cfd6d55d33bd8869fad173040
```
Comment (skyclouds2001) [inline files/zh-cn/web/api/barcodedetector/barcodedetector/index.md:5]:
```suggestion
  sourceCommit: 78d53558b704be923e00aa2664f47a93c32652b4
```

File: files/zh-cn/web/api/barcodedetector/barcodedetector/index.md [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+```
+
+### 参数
+
+- `options` {{optional_inline}}
+
+  - : 包含一系列 `BarcodeFormats` 的选项对象，用于在后续调用 {{domxref('BarcodeDetector.detect()','detect()')}} 中搜索。选项有：
+
+    - `formats` {{optional_inline}}
+      - : 一个表示条形码格式的字符串 {{jsxref('Array')}}。如果未提供，调用 `detect()` 方法时将搜索所有支持的格式。因此，出于性能原因，建议限制为具体的格式。要查看支持格式的完整列表，请参阅 {{domxref("Barcode Detection API", "", "", "nocode")}}。
```
Comment (skyclouds2001) [inline files/zh-cn/web/api/barcodedetector/barcodedetector/index.md:26]:
```suggestion
      - : 一个表示条形码格式的字符串 {{jsxref('Array')}}。如果未提供，调用 `detect()` 方法时将搜索所有支持的格式。因此，出于性能原因，建议限制为具体的格式。要查看支持格式的完整列表，请参阅[受支持的条形码格式类型](/zh-CN/docs/Web/API/Barcode_Detection_API#支持的条形码格式)。
```

File: files/zh-cn/web/api/barcodedetector/barcodedetector/index.md (hunk 4 above)
Comment (skyclouds2001) [inline files/zh-cn/web/api/barcodedetector/barcodedetector/index.md:10]:
```suggestion
**`BarcodeDetector()`** 构造函数创建一个新的用于检测图像中条形码和二维码的 {{domxref("BarcodeDetector")}} 对象。
```

**Review by yin1999 — APPROVED** [review]
LGTM. Thanks @skyclouds2001

**Pull request merged**